src/tests/src/solver/modelParser/testConvertorVisitor.cpp

```diff
@@ -0,0 +1,200 @@
+#include "antares/solver/expressions/visitors/AstDOTStyleVisitor.h"
+#include "antares/solver/expressions/visitors/CompareVisitor.h"
+#include "antares/solver/libObjectModel/library.h"
+#include "antares/solver/modelConverter/convertorVisitor.h"
+#include "antares/solver/modelConverter/modelConverter.h"
+#include "antares/solver/modelParser/Library.h"
+
+using namespace Antares::Solver;
+
+struct Fixture
+{
+    ModelParser::Model model;
+    Antares::Solver::Registry<Antares::Solver::Nodes::Node> registry;
+};
+
+static Nodes::LiteralNode* toLiteral(Nodes::Node* n)
+{
+    return dynamic_cast<Nodes::LiteralNode*>(n);
+}
+
+BOOST_FIXTURE_TEST_CASE(empty_expression, Fixture)
+{
+    auto* node = ModelConverter::convertExpressionToNode("", registry, model);
+    BOOST_CHECK_EQUAL(node, nullptr);
+}
+
+BOOST_FIXTURE_TEST_CASE(negation, Fixture)
+{
+    std::string expression = "-7";
+    auto* n = ModelConverter::convertExpressionToNode(expression, registry, model);
+    BOOST_CHECK_EQUAL(n->name(), "NegationNode");
+    auto* nodeNeg = dynamic_cast<Nodes::NegationNode*>(n);
+    BOOST_CHECK_EQUAL(toLiteral(nodeNeg->child())->value(), 7);
+}
+
+BOOST_FIXTURE_TEST_CASE(identifier, Fixture)
+{
+    ModelParser::Library library;
+    ModelParser::Model model0{
+      .id = "model0",
+      .description = "description",
+      .parameters = {{"param1", true, false}, {"param2", false, false}},
+      .variables = {{"varP", "7", "pmin", ModelParser::ValueType::CONTINUOUS}},
+      .ports = {},
+      .port_field_definitions = {},
+      .constraints = {},
+      .objective = "objectives"};
+
+    std::string expression = "param1";
+    auto* n = ModelConverter::convertExpressionToNode(expression, registry, model0);
+    BOOST_CHECK_EQUAL(n->name(), "ParameterNode");
+
+    expression = "varP";
+    n = ModelConverter::convertExpressionToNode(expression, registry, model0);
+    BOOST_CHECK_EQUAL(n->name(), "VariableNode");
+}
+
+bool expectedMessage(const std::exception& ex)
+{
+    BOOST_CHECK_EQUAL(ex.what(),
+                      std::string("No parameter or variable found for this identifier: abc"));
+    return true;
+}
+
+BOOST_FIXTURE_TEST_CASE(identifierNotFound, Fixture)
+{
+    ModelParser::Library library;
+    ModelParser::Model model0{
+      .id = "model0",
+      .description = "description",
+      .parameters = {{"param1", true, false}, {"param2", false, false}},
+      .variables = {{"varP", "7", "pmin", ModelParser::ValueType::CONTINUOUS},
+                    {"var2", "0", "param2", ModelParser::ValueType::CONTINUOUS}},
+      .ports = {},
+      .port_field_definitions = {},
+      .constraints = {},
+      .objective = "objectives"};
+
+    std::string expression = "abc"; // not a param or var
+    BOOST_CHECK_EXCEPTION(ModelConverter::convertExpressionToNode(expression, registry, model0),
+                          std::exception,
+                          expectedMessage);
+}
+
+BOOST_FIXTURE_TEST_CASE(AddSub, Fixture)
```

Comment:
There are 2 unit tests here.
Making 2 unit tests instead of one would allow to be more specific and clear in the name of tests